src/nemos/basis/_basis_mixin.py

```diff
@@ -0,0 +1,260 @@
+"""Mixin classes for basis."""
+
+from __future__ import annotations
+
+import copy
+import inspect
+from typing import Optional, Tuple, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+from ..convolve import create_convolutional_predictor
+from ._transformer_basis import TransformerBasis
+
+
+class EvalBasisMixin:
+    """Mixin class for evaluational basis."""
+
+    def __init__(self, bounds: Optional[Tuple[float, float]] = None):
+        self.bounds = bounds
+
+    def _compute_features(self, *xi: ArrayLike):
+        """
+        Apply the basis transformation to the input data.
+
+        The basis evaluated at the samples, or :math:`b_i(*xi)`, where :math:`b_i` is a
```

Comment:
This is really abstract, should be made more concrete. Maybe saying something like "The basis is evaluated at the locations specified in the inputs. For example, `basis.compute_features([0, .5])` would return the array:

```
b_1(0) ... b_n(0)
b_1(.5) ... b_n(.5)
```

where `n` is `n_basis_funcs`
"

or something like that